cleverhans/attacks_tf.py

```diff
@@ -583,23 +583,27 @@ def jacobian_augmentation(sess, x, X_sub_prev, Y_sub, grads, lmbda,
     # Create new numpy array for adversary training data
     # with twice as many components on the first dimension.
     X_sub = np.vstack([X_sub_prev, X_sub_prev])
-
-    # For each input in the previous' substitute training iteration
-    for ind, prev_input in enumerate(X_sub_prev):
-        # Select gradient corresponding to the label predicted by the oracle
-        grad = grads[Y_sub[ind]]
-
-        # Prepare feeding dictionary
-        feed_dict = {x: np.reshape(prev_input, input_shape)}
+    nsamples = X_sub_prev.shape[0]
+
+    # Creating and processing as batch
+    nb_batches_aug = int((len(X_sub_prev) + aug_batch_size - 1)/aug_batch_size)
+    p_idxs = 0
+    for i in range(nb_batches_aug):
+        X_batch = X_sub_prev[i*aug_batch_size:(i+1)*aug_batch_size, :, :, :]
```

Comment:
The data is not necessarily shaped as an image in 4 dims, can you use one `:` instead to allow for arbitrary shapes?

Reply:
I think using `...`  is better as this makes the code reader known that its multi-dimensional and not just 2.